packages/new-editor/src/components/editor/data/Variable.test.ts

```diff
@@ -0,0 +1,35 @@
+import { getSelectedMetadataOption, metadataOptions, type Variable } from './Variable';
+
+const variable: Variable = {
+  description: '',
+  metadata: '',
+  name: '',
+  value: '',
+  children: []
+};
+
+describe('Variable', () => {
```

Comment:
Technically totally fine test. But IMO useless if we type the Variable correctly ;)
What if we could type metadata like it only accepts a value which matches on of the possible option values?
Then we can let typescript do the check that we don't put something wrong in it :)

https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgNUygUwCMAbPAbwCg44ATYAZ22LBgQiQC45mYxJAHMA3HTgBbYDEwNMsvjACeYYBHRSZchZgDyHLkmYBtJAFdJpNAF0TAcgBumcueD2b4gL40aoSLBw2NwCWrLysgacIXAAvHAmElRw5GTA5Hz2AHLc7gA0cM6uwJlIufZwXnlJKWkZcPYACpjMzADu0Az2BUVumWAt7Z0VVTWp1vX2ACKYypzS3YUufQ3ycwgLldX0yePpmQCiFpKLvSUNwMcj23C7dZkAYgiUp8vn9ujP7pU0NnAtQRC8GYCgQzE+LDgAEEoFBZgAeO4TPgCIRiJbFFGCZDCSoAPnENGCxngFEwfEIxDIlDitwkTFY7GivAazAg0hgAAscd0JBydIoGuYkABrMptJD2GheIA
```ts
export interface Variable {
  description: string;
  metadata: typeof metadataOptions[number]['value'];
}

export const metadataOptions = [
  { label: 'None', value: 'none' },
  { label: 'Password', value: 'password' },
  { label: 'Daytime', value: 'daytime' },
  { label: 'Enum', value: 'enum' },
  { label: 'File', value: 'file' }
] as const satisfies Array<{ label: string; value: string }>;

const bla: Variable = {
  description: 'something',
  metadata: 'unknown'
}
```

Reply:
I am gonna implement this change. But I think the only test that is unnecessary then is the one where a value is supplied that is not present in `metadataOptions` since this is no longer possible. I still want to know that all the possible values are present in `metadataOptions` and that `getSelectedMetadataOption` returns the correct one of those, right?

Reply:
yes technically this is true.
But I quickly checked this method at all. Why it is even needed?
you use it currently here: https://github.com/axonivy/config-editor-client/blob/446bb138deee91c90412dd50553682aaa4ad8a12/packages/new-editor/src/components/editor/variables/detail/Variable.tsx#L4
To set the value to the select component.
Why can you not simply set the value of the variable to the select?
```tsx
<SimpleSelect value={variable.metadata} items={metadataOptions} onValueChange={value => handleVariableAttributeChange('metadata', value)} />
```